front-end/studio/src/app/services/template.service.ts

```diff
@@ -17,1662 +17,55 @@
 
 import {Injectable} from "@angular/core";
 import {ApiDesignTemplate} from "../models/api-design-template.model";
+import {AbstractHubService} from "./hub";
+import {HttpClient} from "@angular/common/http";
+import {IAuthenticationService} from "./auth.service";
+import {ConfigService} from "./config.service";
 
 
 /**
  * A service to manage a list of API Design templates that can be used when creating new
  * APIs.
  */
 @Injectable()
-export class TemplateService {
+export class TemplateService extends AbstractHubService {
 
-    private templates: ApiDesignTemplate[];
+    private templateCache: { [key: string]: ApiDesignTemplate[] } = {};
 
-    constructor() {
-        this.templates = this.loadTemplates();
-    }
-
-    private loadTemplates(): ApiDesignTemplate[] {
-        return [
-            {
-                type: "OpenAPI20",
-                name: "Simple API",
-                description: "Creates a very simple API with some basic settings that can be used as a starting point for your custom API.",
-                content: SIMPLE_20
-            },
-            {
-                type: "OpenAPI20",
-                name: "Pet Store Example",
-                description: "Creates a standard/famous Swagger 2.0 Pet Store API sample.  Can be found here: https://petstore.swagger.io/v2/swagger.json",
-                content: PET_STORE_20
-            },
-            {
-                type: "OpenAPI30",
-                name: "Pet Store Example",
-                description: "Creates a version of the classic Swagger Pet Store example but redesigned using OpenAPI 3!",
-                content: PET_STORE_30
-            },
-            {
-                type: "OpenAPI30",
-                name: "USPTO Dataset API",
-                description: "Creates an API using the USPTO Data Set API (DSAPI) as a basis.",
-                content: USPTO_30
-            },
-            {
-                type: "AsyncAPI20",
-                name: "Signed-up Event API",
-                description: "Creates an API that send user signed events using the AsyncAPI 2.0 standard",
-                content: USER_SIGNEDUP_20
-            }
-        ];
+    /**
+     * Constructor.
+     * @param http
+     * @param authService
+     * @param config
+     */
+    constructor(http: HttpClient, authService: IAuthenticationService, config: ConfigService) {
+        super(http, authService, config);
     }
 
     /**
      * Gets a list of templates for the given spec version.
      * @param type
      */
     public getTemplates(type: string): ApiDesignTemplate[] {
```

Comment:
I think this needs to return a `Promise<ApiDesignTemplate[]>` if we're going to possibly be fetching the data from the server now (when not cached).